x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -7,64 +7,162 @@
+import { CustomUrlAnomalyRecordDoc, KibanaUrlConfig } from '../../../../common/types/custom_urls';
+import { TimeRangeBounds } from '../../util/time_buckets';
+import { AnomalyRecordDoc } from '../../../../common/types/anomalies';
+import { MlKibanaReactContextValue } from '../../contexts/kibana';
+// @ts-ignore
+import { getFieldTypeFromMapping } from '../../services/mapping_service';
+import { useFieldFormatter } from '../../contexts/kibana/use_field_formatter';
+import { FIELD_FORMAT_IDS } from '../../../../../../../src/plugins/field_formats/common';
+
+interface Anomaly {
+  jobId: string;
+  customUrls: KibanaUrlConfig[];
+  source: AnomalyRecordDoc;
+  entityValue: string;
+  entityName: string;
+  isTimeSeriesViewRecord: boolean;
+  time: number;
+}
+interface LinksMenuProps {
+  anomaly: Anomaly;
+  bounds: TimeRangeBounds;
+  showViewSeriesLink: boolean;
+  isAggregatedData: boolean;
+  interval: 'day' | 'hour' | 'second';
+  showRuleEditorFlyout: (anomaly: Anomaly) => void;
+  kibana: MlKibanaReactContextValue;
+}
+
+export const LinksMenuUI = (props: LinksMenuProps) => {
+  const [isPopoverOpen, setPopoverOpen] = useState(false);
+  const [openInDiscoverUrl, setOpenInDiscoverUrl] = useState<string | undefined>();
+
+  const dateFormatter = useFieldFormatter(FIELD_FORMAT_IDS.DATE);
+
+  useEffect(() => {
+    let unmounted = false;
+    const generateDiscoverUrl = async () => {
+      const {
+        services: { share },
+      } = props.kibana;
+      const discoverLocator = share.url.locators.get('DISCOVER_APP_LOCATOR');
+
+      if (!discoverLocator) {
+        // eslint-disable-next-line no-console
+        console.error('No locator for Discover detected');
+        return;
+      }
+
+      const job = mlJobService.getJob(props.anomaly.jobId);
+
+      const index = job.datafeed_config.indices[0];
+      const interval = props.interval;
+      const dataViewId = (await getDataViewIdFromName(index)) || index;
+      const record = props.anomaly.source;
+
+      const earliestMoment = moment(record.timestamp).startOf(interval);
+      if (interval === 'hour') {
+        // Start from the previous hour.
+        earliestMoment.subtract(1, 'h');
+      }
+      let latestMoment = moment(record.timestamp).add(record.bucket_span, 's');
+      if (props.isAggregatedData === true) {
+        latestMoment = moment(record.timestamp).endOf(interval);
+        if (interval === 'hour') {
+          // Show to the end of the next hour.
+          latestMoment.add(1, 'h'); // e.g. 2016-02-08T18:59:59.999Z
+        }
+      }
+      const from = dateFormatter(earliestMoment.unix() * 1000); // e.g. 2016-02-08T16:00:00.000Z
+      const to = dateFormatter(latestMoment.unix() * 1000);
+
+      let kqlQuery = '';
+
+      if (record.influencers) {
+        kqlQuery = record.influencers
+          // No need to add categorization detectors to query
+          .filter((influencer) => influencer.influencer_field_name !== 'mlcategory')
```

Comment:
I think if we are showing 'View examples' in the menu, then there is no point adding in 'View in Discover', as the 'View examples' opens up Discover anyway, with the query set to show docs for the anomalous category. I don't think 'View in Discover' showing the unfiltered docs adds any value to the existing 'View examples'. You should just check for `anomaly.entityName === 'mlcategory'` and not add View in Discover in this case.

![image](https://user-images.githubusercontent.com/7405507/145835463-d6462eb2-d4d4-4a03-a206-33977c183d34.png)

Reply:
Might be worth adding some sort of `isCategorizationAnomaly` function to `ml/common/util/anomaly_utils.ts`